omod/src/main/java/org/openmrs/module/metrics/util/MetricHandler.java

```diff
@@ -2,44 +2,40 @@
 
 import static com.codahale.metrics.MetricRegistry.name;
 
-import javax.management.InstanceAlreadyExistsException;
-import javax.management.MBeanRegistrationException;
-import javax.management.MBeanServer;
 import javax.management.MalformedObjectNameException;
-import javax.management.NotCompliantMBeanException;
 import javax.management.ObjectName;
-import java.lang.management.ManagementFactory;
+import javax.servlet.http.HttpServletRequest;
 import java.time.LocalDateTime;
 import java.util.Map;
 
 import com.codahale.metrics.MetricRegistry;
-import com.codahale.metrics.jmx.JmxReporter;
 import com.codahale.metrics.jvm.JmxAttributeGauge;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
 import org.openmrs.module.metrics.api.exceptions.MetricsException;
 import org.openmrs.module.metrics.api.service.MetricService;
-import org.openmrs.module.metrics.api.utils.EventsUtils;
 import org.openmrs.module.metrics.builder.JmxReportBuilder;
 import org.openmrs.module.metrics.model.MetricscConfigImpl;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 
 public class MetricHandler {
```

Comment:
Instead of converting all of this to `static` methods, I would make this a Spring `@Component` and inject a `MetricHandler` instance into the servlet. Spring can't properly inject `static` fields.

Basically, just change the start to:

```
@Component
public class MetricHandler {
```

Reply:
Sure